gef-ui/server/controllers/amendments/cover-end-date/get-cover-end-date.ts

```diff
@@ -0,0 +1,39 @@
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { Response } from 'express';
+import * as api from '../../../services/api';
+import { CoverEndDateViewModel } from '../../../types/view-models/amendments/cover-end-date-view-model';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { userCanAmendFacility } from '../../../utils/facility-amendments.helper';
+
+export type GetCoverEndDateRequest = CustomExpressRequest<{
+  params: { dealId: string; facilityId: string; amendmentId: string };
+}>;
+
+export const getCoverEndDate = async (req: GetCoverEndDateRequest, res: Response) => {
+  try {
+    const { dealId, facilityId, amendmentId } = req.params;
+    const { userToken, user } = asLoggedInUserSession(req.session);
+
+    const deal = await api.getApplication({ dealId, userToken });
```

Comment:
Please validate `userToken` before making an API call.

Reply:
@abhi-markan could you give me an example please? I followed the same approach as we have in other files. I haven't seen any other different example. Does not asLoggedInUserSession check the session ?